cly/console.py

```diff
@@ -220,7 +220,9 @@ def __init__(self, *args, **kwargs):
     }
 
     def decode(self, input, errors='strict'):
-        return _decode_re.sub(self._decode_match, str(input, encoding='ascii'))
+        if not isinstance(input, str):
```

Comment:
```suggestion
        if isinstance(input, str):
            input = str
        elif isinstance(input, bytes):
            input = str(input, encoding='ascii')
        else:
            AttributeException...
```

Reply:
This is all very confusing... while trying to understand the code, I just realized that `str` is not the real `str` (at least in Python 2) but some imported wrapper (since `str("foo", encoding='ascii')` is not valid in Python 2).

*sigh*

I can't really dive into this deeper; I suggest doing what is safest even if it's a longer/uglier -- worst case, put the conversion in a separate function if it's used often.

Reply:
I will change into @danni-m 's option